### PR TITLE
📝 Add docs for OTel header generation and ingestion

### PR DESCRIPTION
**Pull request by fuzzybinary**
### What and why?

Add documentation on how to generate OTel and `traceparent` style headers.

Also add a bit of troubleshooting documentation.

### Review checklist

- [ ] This pull request has appropriate unit and / or integration tests 
- [ ] This pull request references a Github or JIRA issue

### CI Configuration (optional)
- [ ] Run unit tests
- [ ] Run integration tests

### CODE REVIEW

**Comment by alai97** [comment]
Opened https://datadoghq.atlassian.net/browse/DOCS-4654 for an editorial review!

File: packages/datadog_flutter_plugin/doc/rum/otel_support.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# RUM Flutter Open Telemetry Support
+
+The [Datadog Tracking Http Client][1] package and [gRPC Interceptor][2] package both support distributed traces through both automatic header generation and header ingestion.
```
Comment (plousada) [inline packages/datadog_flutter_plugin/doc/rum/otel_support.md:3]: Header ingestion?

File: packages/datadog_flutter_plugin/doc/troubleshooting.md [hunk 2]
```diff
@@ -30,6 +30,20 @@ DatadogSdk.instance.sdkVerbosity = Verbosity.verbose;
...
+
+The most common reason users aren't seeing errors in RUM is because there is no view started. Make sure you are staring a view with `DatadogSdk.instance.rum?.startView` or that if you are using `DatadogRouteObserver` that your current Route has a name.
+
+## Not seeing Distributed Traces on Resources
+
+There are three possible settings you should check related to distributed tracing.  
+
+First, if you are not seeing any resources loads for your RUM View, check that you have setup either the `DatadogTrackingHttpClient` with `enableHttpTracking`, or that you are properly wrapping a `Client` from the `http` package with `DatadogClient`. 
+
+Second, ensure that you have properly set `firstPartyHosts` in your configuration. Datadog only generates and sends tracing headers to the hosts you specify in this property. Note that `firstPartyHosts` does not support wildcards or Regular Expressions.
```
Comment (plousada) [inline packages/datadog_flutter_plugin/doc/troubleshooting.md:43]: We support wildcard in Android. On iOS I think not. 
We might want to align this.

**Review by plousada — APPROVED** [review]

File: packages/datadog_flutter_plugin/doc/rum/otel_support.md (hunk 1 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/doc/rum/otel_support.md:3]: Good catch, I wrote this before I removed the ability for the client to parse and forward OTel headers.

File: packages/datadog_flutter_plugin/doc/troubleshooting.md (hunk 2 above)
Comment (fuzzybinary) [reply packages/datadog_flutter_plugin/doc/troubleshooting.md:43]: Yes we likely want to align, but in this case Flutter is handling (almost) all the logic for doing host matching, and I didn't code in the logic for REs or wildcards.

File: packages/datadog_flutter_plugin/doc/rum/otel_support.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+// For default Datadog HTTP tracing:
+final configuration = DdSdkConfiguration(
+    // configuration
+    firstPartyHostsWithTracingHeaders: hostHeaders,
+);
+```
+
+Then enable tracing as normal.
+
+Note that this information will be merged with any hosts set on `DdSdkConfiguration.firstPartyHosts`. Hosts specified in `firstPartyHosts` will generate Datadog Tracing Headers by default.
```
Comment (rtrieu) [inline packages/datadog_flutter_plugin/doc/rum/otel_support.md:28]:
```suggestion
This information is merged with any hosts set on `DdSdkConfiguration.firstPartyHosts`. Hosts specified in `firstPartyHosts` generate Datadog Tracing Headers by default.
```

File: packages/datadog_flutter_plugin/doc/troubleshooting.md [hunk 4]
```diff
@@ -30,6 +30,20 @@ DatadogSdk.instance.sdkVerbosity = Verbosity.verbose;
 
 This causes the SDK to output additional information about what it's doing and what errors it's encountering, which may help you and Datadog Support narrow down your issue.
 
+## Not seeing Errors
+
+The most common reason users aren't seeing errors in RUM is because there is no view started. Make sure you are staring a view with `DatadogSdk.instance.rum?.startView` or that if you are using `DatadogRouteObserver` that your current Route has a name.
```
Comment (rtrieu) [inline packages/datadog_flutter_plugin/doc/troubleshooting.md:35]:
```suggestion
If you do not see any errors in RUM, it's likely no view has been started. Make sure you have started a view with `DatadogSdk.instance.rum?.startView` or, if you are using `DatadogRouteObserver` make sure your current Route has a name.
```

File: packages/datadog_flutter_plugin/doc/troubleshooting.md [hunk 5]
```diff
@@ -30,6 +30,20 @@ DatadogSdk.instance.sdkVerbosity = Verbosity.verbose;
...
+
+## Not seeing Distributed Traces on Resources
+
+There are three possible settings you should check related to distributed tracing.  
+
+First, if you are not seeing any resources loads for your RUM View, check that you have setup either the `DatadogTrackingHttpClient` with `enableHttpTracking`, or that you are properly wrapping a `Client` from the `http` package with `DatadogClient`.
```
Comment (rtrieu) [inline packages/datadog_flutter_plugin/doc/troubleshooting.md:41]:
```suggestion
If you are not seeing any resources load for your RUM view, first check that you have set up either the `DatadogTrackingHttpClient` with `enableHttpTracking`, or that you are properly wrapping a `Client` from the `http` package with `DatadogClient`. 
```

File: packages/datadog_flutter_plugin/doc/troubleshooting.md (hunk 2 above)
Comment (rtrieu) [inline packages/datadog_flutter_plugin/doc/troubleshooting.md:43]:
```suggestion
Second, ensure that you have properly set `firstPartyHosts` in your configuration. Datadog only generates and sends tracing headers to the hosts you specify in this property. `firstPartyHosts` does not support wildcards or Regular Expressions.
```

File: packages/datadog_flutter_plugin/doc/troubleshooting.md [hunk 6]
```diff
@@ -30,6 +30,20 @@ DatadogSdk.instance.sdkVerbosity = Verbosity.verbose;
...
+
+## Not seeing Distributed Traces on Resources
+
+There are three possible settings you should check related to distributed tracing.  
+
+First, if you are not seeing any resources loads for your RUM View, check that you have setup either the `DatadogTrackingHttpClient` with `enableHttpTracking`, or that you are properly wrapping a `Client` from the `http` package with `DatadogClient`. 
+
+Second, ensure that you have properly set `firstPartyHosts` in your configuration. Datadog only generates and sends tracing headers to the hosts you specify in this property. Note that `firstPartyHosts` does not support wildcards or Regular Expressions.
+
+Lastly, Datadog sets your tracing sample rate to 20% by default, so distributed traces will only be available on 20% of resource loads. While debugging, you may want to set `RumConfiguration.tracingSamplingRate` property to 100.
```
Comment (rtrieu) [inline packages/datadog_flutter_plugin/doc/troubleshooting.md:45]:
```suggestion
Lastly, Datadog sets your tracing sample rate to 20% by default, so distributed traces are only available for 20% of resource loads. While debugging, you may want to set `RumConfiguration.tracingSamplingRate` property to 100.
```

**Review by rtrieu — CHANGES_REQUESTED** [review]
Left some edit suggestions! Otherwise, it looks good.

**Review by maciejburda — APPROVED** [review]

**Pull request merged**